tests/rules/no-reference-like-url.test.js

```diff
@@ -0,0 +1,295 @@
+/**
+ * @fileoverview Tests for no-reference-like-url rule.
+ * @author TKDev7
+ */
+
+//------------------------------------------------------------------------------
+// Imports
+//------------------------------------------------------------------------------
+
+import rule from "../../src/rules/no-reference-like-url.js";
+import markdown from "../../src/index.js";
+import { RuleTester } from "eslint";
+import dedent from "dedent";
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+	plugins: {
+		markdown,
+	},
+	language: "markdown/commonmark",
+});
+
+ruleTester.run("no-reference-like-url", rule, {
```

Comment:
Let's add  valid/invalid test cases for:

- Links/images with titles.
- Identifiers with whitespace or special characters.
- Link with whitespaces:
```md
[Click Me
](test)

[test]: https://abc.com
```